index.js

```diff
@@ -123,26 +126,28 @@ ExpressOAuthServer.prototype.token = function(options) {
 
 var handleResponse = function(req, res, response) {
   res.set(response.headers);
-
-  res.send(response.body);
-  res.status(response.status);
+  res.status(response.status).send(response.body);
 };
 
 /**
  * Handle error.
  */
 
-var handleError = function(e, req, res, response) {
-  if (response) {
-    res.set(response.headers);
-  }
+var handleError = function(e, req, res, response, next) {
 
-  if (e instanceof UnauthorizedRequestError) {
-    return res.status(e.code);
-  }
+  if (this.useErrorHandler === true) {
```

Comment:
Oh, I just realized what you're talking about. The problem is that `this` is already wrong by the time `handleError.call(this, ...)` is called. `Promise.bind(this)` is supposed to help with that, but `this` refers to the returned middleware.

This can be solved by changing

```
var server = this.server;
```
to
```
var that = this;
```

and

```
return Promise.bind(this)
```
to
```
return Promise.bind(that)
```

in all 3 handlers.

Obviously references to `server` have to be replaced by `this.server` after this change.